Phase03/Program.cs

```diff
@@ -0,0 +1,39 @@
+﻿
+using Newtonsoft.Json;
+class Program
+{
+    private static List<StudentData> students;
+    private static List<StudentGrade> grades;
+    static void Main()
+    {
+        InitializeLists();
+        FindAverageScores();
+        PrintResult(GetSortedAverages());
+    }
+    private static void InitializeLists()
+    {
+        students = GetDataFromJson<StudentData>("data/students.json");
+        grades = GetDataFromJson<StudentGrade>("data/grades.json");
+    }
+    private static List<T> GetDataFromJson<T>(string filePath)
+    {
+        using (StreamReader fileReader = new StreamReader(filePath))
+        {
+            var json = fileReader.ReadToEnd();
+            return JsonConvert.DeserializeObject<List<T>>(json);
+        }
+    }
+    private static void FindAverageScores()
+    {
+        students.ForEach(student => student.Average = grades.Where(
+            o => o.StudentNumber == student.StudentNumber).Average(g => g.Score));
```

Comment:
These two lines are a little complicated and hard to read.

Reply:
This is a usage of _linq_ with _lambda expression_. 
For each student, it sets the average score of that student. 
How does it find the average score? Each student has a student number. So does each grade. The code "grades.Where(..." finds the grades that have the same student number as the student. Then the average value of those grades is set as the average score of the student.